src/parameterized_tendencies/radiation/RRTMGPInterface.jl

```diff
@@ -805,17 +805,44 @@ function RRTMGPModel(
         end
 
         if radiation_mode.aerosol_radiation
-            aero_type = DA{Int}(undef, nlay, ncol)
-            name = "center_aerosol_type"
+            n_aerosols = 7
+            aero_mask = DA{Bool}(undef, nlay, ncol)
+            name = "center_aerosol_mask"
             set_and_save!(aero_type, name, t..., dict)
-            aero_size = DA{FT}(undef, nlay, ncol)
-            name = "center_aerosol_radius"
-            set_and_save!(aero_size, name, t..., dict)
-            aero_mass = DA{FT}(undef, nlay, ncol)
-            name = "center_aerosol_column_mass_density"
-            set_and_save!(aero_mass, name, t..., dict)
+
+            aero_size = DA{FT}(undef, n_aerosols, nlay, ncol)
+            dust_aero_size = view(aero_size, 1, :, :)
+            ss_aero_size = view(aero_size, 2, :, :)
+            sulfate_aero_size = view(aero_size, 3, :, :)
+            bcpi_aero_size = view(aero_size, 4, :, :)
+            bcpo_aero_size = view(aero_size, 5, :, :)
+            ocpi_aero_size = view(aero_size, 6, :, :)
+            ocpo_aero_size = view(aero_size, 7, :, :)
+            set_and_save!(dust_aero_size, "center_dust_radius", t..., dict)
+            set_and_save!(ss_aero_size, "center_ss_radius", t..., dict)
+            set_and_save!(sulfate_aero_size, "center_sulfate_radius", t..., dict)
+            set_and_save!(bcpi_aero_size, "center_bcpi_radius", t..., dict)
+            set_and_save!(bcpo_aero_size, "center_bcpo_radius", t..., dict)
+            set_and_save!(ocpi_aero_size, "center_ocpi_radius", t..., dict)
+            set_and_save!(ocpo_aero_size, "center_ocpo_radius", t..., dict)
+
+            aero_mass = DA{FT}(undef, n_aerosols, nlay, ncol)
+            dust_aero_mass = view(aero_mass, 1, :, :)
+            ss_aero_mass = view(aero_mass, 2, :, :)
+            sulfate_aero_mass = view(aero_mass, 3, :, :)
+            bcpi_aero_mass = view(aero_mass, 4, :, :)
+            bcpo_aero_mass = view(aero_mass, 5, :, :)
+            ocpi_aero_mass = view(aero_mass, 6, :, :)
+            ocpo_aero_mass = view(aero_mass, 7, :, :)
+            set_and_save!(dust_aero_mass, "center_dust_column_mass_density", t..., dict)
+            set_and_save!(ss_aero_mass, "center_ss_column_mass_density", t..., dict)
+            set_and_save!(sulfate_aero_mass, "center_sulfate_column_mass_density", t..., dict)
+            set_and_save!(bcpi_aero_mass, "center_bcpi_column_mass_density", t..., dict)
+            set_and_save!(bcpo_aero_mass, "center_bcpo_column_mass_density", t..., dict)
+            set_and_save!(ocpi_aero_mass, "center_ocpi_column_mass_density", t..., dict)
+            set_and_save!(ocpo_aero_mass, "center_ocpo_column_mass_density", t..., dict)
```

Comment:
```suggestion
aero_size = DA{FT}(undef, n_aerosols, nlay, ncol)
aero_mass = DA{FT}(undef, n_aerosols, nlay, ncol)

aerosol_names = [
    "dust", "ss", "sulfate", "bcpi", "bcpo", "ocpi", "ocpo"
]

for (i, name) in enumerate(aerosol_names)
    set_and_save!(view(aero_size, i, :, :), "center_$(name)_radius", t..., dict)
    set_and_save!(view(aero_mass, i, :, :), "center_$(name)_column_mass_density", t..., dict)
end
```

Reply:
I think

Reply:
Thanks! I was going to test how to make it simpler.